app/models/rental.rb

```diff
@@ -8,18 +8,24 @@ class Rental < ActiveRecord::Base
   has_many :financial_transactions
   has_one :financial_transaction, as: :transactable
 
-  after_create :create_financial_transaction
+  # will skip financial transactions if we plan to create manual pricing
+  attr_accessor :skip_financial_transactions
```

Comment:
Is there the possibility of a race condition with this?

Reply:
Could you provide an example?  I am not certain I understand.

Reply:
Your code changes the class-wide attr_accessor, and another application instance accesses the DB at the same time.

Reply:
oh shoot, I thought attr_accessors values were instance variables

Reply:
They are. I'm just curious if they're model wide like that, so that while your code has it set one way or the other, it could affect other people interacting with the model. I'm honestly not sure.

Reply:
Could you possibly give an example, I am not sure I understand what you're saying.

Reply:
The more I think about it, the more I suspect it's not a concern. But presumably you're running this server with something like Phusion Passenger, which makes available multiple running instances of your app so that you can serve multiple requests simultaneously. So, my concern was that while one instance of the app was processing a request which involved setting the model-wide `attr_accessor` to skip financial transactions, another app instance responding to another, unrelated request involving the same model might also skip financial transactions.

But, I think since it's a separate instance of the app running, and that `attr_accessor` is essentially a value in the running memory of a single instance, that it's not a real concern, since a single app instance can't serve multiple requests at once.

@werebus, is that correct?

Reply:
You are correct, I don't think it's a concern.

Although they are the same _database_ object, it's very unlikely (impossible? probably) that you have two references to the same _ruby_ object.